src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -646,9 +671,9 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, Diagnostic
             }
         }
 
-        private static BoundExpression BindDefaultLiteral(ExpressionSyntax node)
+        private static BoundDefaultLiteral BindDefaultLiteral(ExpressionSyntax node)
         {
-            return new BoundDefaultExpression(node, type: null);
+            return new BoundDefaultLiteral(node, type: null);
```

Comment:
>, type: null [](start = 47, length = 12)

I think there is a way to adjust description of this node in BoundNodes.xml to prevent consumers from passing a type.